src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/LanguageClient/ProjectSystem/RazorProjectInfoEndpointPublisher.TestAccessor.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.VisualStudio.LanguageServer.ContainedLanguage;
+
+namespace Microsoft.VisualStudio.Razor.LanguageClient.ProjectSystem;
+
+internal partial class RazorProjectInfoEndpointPublisher
+{
+    internal TestAccessor GetTestAccessor() => new(this);
+
+    [SuppressMessage("RoslynDiagnosticsMaintainability", "RS0043:Do not call 'GetTestAccessor()'", Justification = "Used by test code only")]
+    internal static TestAccessor CreateTestInstanceAccessor(LSPRequestInvoker requestInvoker, IProjectSnapshotManager projectSnapshotManager)
+        => new(new RazorProjectInfoEndpointPublisher(requestInvoker, projectSnapshotManager, TimeSpan.FromMilliseconds(1)));
+
+    internal sealed class TestAccessor(RazorProjectInfoEndpointPublisher instance)
+    {
+        /// <summary>
+        /// Allows unit tests to imitate ProjectManager.Changed event firing
+        /// </summary>
+        public void ProjectManager_Changed(object sender, ProjectChangeEventArgs args)
+            => instance.ProjectManager_Changed(sender, args);
```

Comment:
How annoying is it to not expose this, but instead have the tests use the project spanshot manager and just push updates? Unless we have tests about abnormal event ordering, it worries that this could mean our tests aren't validating real world behaviour if/when the project snapshot manager changes

Reply:
I pinged you offline, but basically for scheduling purposes, I'd rather do that as a separate PR. In general it seems OK to me since these are unit tests rather than integration tests, though I don't disagree with your suggestions. Please let me know what you think. Thanks!

Reply:
Per our offline conversation, I'll leave it as is for now. We can tweak it later if the code actually goes forward.